{
    "cve_id": "CVE-2024-29199",
    "cve_description": "Nautobot is a Network Source of Truth and Network Automation Platform. A number of Nautobot URL endpoints were found to be improperly accessible to unauthenticated (anonymous) users. These endpoints will not disclose any Nautobot data to an unauthenticated user unless the Nautobot configuration variable EXEMPT_VIEW_PERMISSIONS is changed from its default value (an empty list) to permit access to specific data by unauthenticated users. This vulnerability is fixed in 1.6.16 and 2.1.9.",
    "cve_publish_date": "2024-03-26",
    "cwe_id": "NVD-CWE-noinfo",
    "commit_message": "View authentication and permission fixes (#5464)",
    "type_of_change": "Modification",
    "filename_of_changes": "views.py",
    "code_language": "Python",
    "number_of_lines_added_for_mitigation": "9",
    "number_of_lines_deleted_vulnerable_to_cve": "9",
    "vulnerable_lines": [
        "// Line_Reference 778:                 return redirect(self.get_return_url(request), default_return_url=\"ipam:ipaddress_add\")",
        "// Line_Reference 861:     default_return_url = \"ipam:ipaddress_add\"",
        "// Line_Reference 864:         # Redirect user if an interface has not been provided",
        "// Line_Reference 865:         if \"interface\" not in request.GET and \"vminterface\" not in request.GET:",
        "// Line_Reference 866:             return redirect(self.get_return_url(request))",
        "// Line_Reference 868:         _, error_msg = retrieve_interface_or_vminterface_from_request(request)",
        "// Line_Reference 869:         if error_msg:",
        "// Line_Reference 870:             messages.warning(request, error_msg)",
        "// Line_Reference 871:             return redirect(self.get_return_url(request))"
    ]
}
